bindings/java/src/org/sleuthkit/datamodel/OsAccount.java

```diff
@@ -245,12 +253,42 @@ public static OsAccountInstanceType fromID(int typeId) {
 	}
 
 	/**
-	 * Sets the account user's full name.
+	 * Set the account login name, such as "jdoe".
+	 * 
+	 * @param loginName Login name to set.
+	 */
+	public void setLoginName(String loginName) {
+		if (this.loginName == null) {
+			this.loginName = loginName;
+			updateSignature();
+			this.isDirty = true;
+		}
+	}
+
+	/**
+	 * Set the account unique id, such as SID or UID
+	 * 
+	 * @param uniqueId Id to set.
+	 */
+	public void setUniqueId(String uniqueId) {
+		if (this.uniqueId == null) {
+			this.uniqueId = uniqueId;
+			updateSignature();
+			this.isDirty = true;
+		}
+	}
+	
+	
+	/**
+	 * Sets the account user's full name, such as "John Doe"
 	 *
 	 * @param fullName Full name.
 	 */
 	public void setFullName(String fullName) {
-		this.fullName = fullName;
+		if (this.fullName == null) {
+			this.fullName = fullName;
+			this.isDirty = true;
+		}
```

Comment:
@bcarrier and @raman-bt, just want to make sure that you have agreed that we want to silently ignore caller input if login name, unique name, etc. have already been set, either by the current caller or some other caller, possibly on a different machine. This could be confusing for the calling code. Alternatives would be to throw an exception or allow overwrites.